Modules/_pickle.c

```diff
@@ -3637,11 +3637,19 @@ save_global(PicklerObject *self, PyObject *obj, PyObject *name)
         goto error;
     }
     if (cls != obj) {
-        Py_DECREF(cls);
-        PyErr_Format(st->PicklingError,
-                     "Can't pickle %R: it's not the same object as %S.%S",
-                     obj, module_name, global_name);
-        goto error;
+        /* bpo-40137: Compare types' name in case some type objects are
+           created in heap. */
```

Comment:
Please move this change into a separated PR.

Reply:
Oh, you are so quick. OK, I create a new one.